hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -228,7 +223,8 @@ public Option<BufferedRecord<T>> deltaMergeNonDeleteRecord(BufferedRecord<T> new
           props);
 
       if (!combinedRecordAndSchemaOpt.isPresent()) {
-        return Option.empty();
+        // An empty Option indicates that the output represents a delete.
+        return Option.of(new BufferedRecord<>(newRecord.getRecordKey(), newRecord.getOrderingValue(), null, null, true));
```

Comment:
When the user indicate a delete through `Option.empty()`, should `DEFAULT_ORDERING_VALUE` be used (as if no ordering value is provided in `Option.empty()`)?  This is to guarantee that the delete is a commit-time-based delete, and if later an insert/update happens after this delete, the insert/update should be treated as insert without considering the "ordering value" of the delete indicated by the `Option.empty()`.  We can make this assumption as the semantics.  If user does want to provide deletes with ordering value, the delete record should be used, which provides the ability to contain the ordering value, different from the semantics of `Option.empty`, so that there is no overlap between these two types of deletes from the custom merger.